### ISSUE TITLE
Support several `redux-offline` instances

### ISSUE THREAD

**Issue by piranna** [redux-offline/redux-offline#314]
In our project, we are using `redux-offline` to manage requests to our own API client object with a custom `effect()` function, and probably we would need to do the same with other services of our own. Obviously this services requires an internet connection, so we have some kind of hierarchy regarding `detectNetwork()` function, one to know if we are online, and another one(s) to know if we are connected or disconnected to our servers (and call `connect()` if needed before being able to do any API call, that would fail if we are disconnected).

We have though about using low level `createOffline()` function with different configs for `detectNetwork()`, `effect()` and friends for each independent service, checking in `detectNetwork()` for the connection to their respective server, and being all of them controlled by a main queue where `detectNetwork()` just only checks that we are online and queue all the `connect()` requests from the services, so they are launched once we get online, and after they get connected to their servers, their own `detectNetwork()` enable them to start processing their own queues.

Unluckily, I've found some limitations on `redux-offline` regarding this point, maybe because it's mostly focused to be used for a single REST web server. For example, to define the store path where the `redux-offline` queue needs to be persisted it seems that [you need to fully replace `offlineStateLens()` function](https://github.com/redux-offline/redux-offline/blob/develop/src/defaults/offlineStateLens.js) instead of just provide a string with the name. That's easy to fix... but the main problem surfaced with internal `redux-offline` events, [that have a hardcoded preffix](https://github.com/redux-offline/redux-offline/blob/develop/src/constants.js), so having several `redux-offline` instance would conflict between them if not using some parameter to identify to whom of them they are part of, that's a bit hackish.

So my question is, is there any way to support several `redux-offline` instances in a single application, all of them using the same store? Or should `redux-offline` to be upgraded to be able of that?

**Comment by piranna** [comment]
Ok, our project has progressed and we have reviewed redux-offline source code, now we have a clearer view of what we need. It's not about having multiple instances of redux-offline in a single app, but instead to have several queues. The way we see it, it's needed to be splitted the redux-persist fuctionality and configuration from the actual network and queue operations. This way, it could be possible to have several queues, each one having its own detectNetwork and effect functions, allowingeach one to work with a different service and control if they are online or not, for example. We have already organized our code towards this and we plan to start with this changes in the next weeks, just only we would like to have some feedback, opinions and sugerences towards this functionality.

**Comment by spruce-bruce** [comment]
I have another use case for multiple queues, one that allows me to divide up requests by priority.

The example I have is when my user logs in I want to prefetch some images so that if they go offline they can view those images. I don't want to add those images to my main redux-offline queue, though, because then my user's network operations will be blocked behind all those images.

It would be nice to load all these low priority image queue requests into their own queue and let them operate in parallel (or when the main queue is idle) so that I get all the benefits of redux-offline's commit/retry/rollback/discard logic

**Comment by spruce-bruce** [comment]
I suppose you can accomplish priority queuing right now by overriding the enqueue function, right? You could enqueue high priority requests before low priority requests

**Comment by piranna** [comment]
Yes, `enqueue()` function is ideal for this, at https://github.com/redux-offline/redux-offline/pull/337 you have one example of that.
